core/src/autogluon/core/scheduler/mo_hyperband.py

```diff
@@ -0,0 +1,170 @@
+    r"""Implements different variants of asynchronous multi-objective Hyperband
+    by extending the standard HyperbandScheduler using random scalarizations. 
+    The implemented method is described in detail in:
+
+        Schmucker, Donini, Perrone, Zafar, Archambeau (2020)
+        Multi-Objective Multi-Fidelity Hyperparameter Optimization with 
+        Application to Fairness
+        https://meta-learn.github.io/2020/papers/24_paper.pdf
+
+    This class supports the full range of parameters of the standard
+    HyperbandScheduler. The 'objectives' replaces the 'reward_attr' parameter.
+    Here we only explain the multi-objective specific parameters. For the full 
+    list of available options refer to HyperbandScheduler.
+
+    Parameters
+    ----------
+    objectives : dict
+        Dictionary with the names of objectives of interest. The corresponding
+        values are allowed to be either "MAX" or "MIN" and indicate if an 
+        objective is to be maximized or minimized.
+    scalarization_options : dict
+        Contains arguments for the underlying scalarization algorithm. 
+        Available algorithms are "random_weights" and "parego".
+
+    Examples
+    --------
+    examples/mo_hpo/mo_hyperband.ipynb
+    """
+
+    def __init__(self, train_fn, **kwargs):
+        self._scalarization_options = kwargs["scalarization_options"]
+        self._objectives = kwargs["objectives"]
+        self._sign_vector = _prepare_sign_vector(self._objectives)
+        kwargs["reward_attr"] = "_SCALARIZATION"
+        super().__init__(train_fn, **filter_by_key(kwargs, _ARGUMENT_KEYS))
+
+    def add_job(self, task, **kwargs):
+        """Adding a training task to the scheduler.
+
+        Args:
+            task (:class:`autogluon.scheduler.Task`): a new training task
+
+        Relevant entries in kwargs:
+        - bracket: HB bracket to be used. Has been sampled in _promote_config
+        - new_config: If True, task starts new config eval, otherwise it 
+          promotes a config (only if type == 'promotion')
+        - elapsed_time: Time stamp
+          Only if new_config == False:
+        - config_key: Internal key for config
+        - resume_from: config promoted from this milestone
+        - milestone: config promoted to this milestone (next from resume_from)
+        - scalarization_options: Specifies scalarazitation method to be used
+        """
+        if "num_weights" in self._scalarization_options:
+            n_weights = self._scalarization_options["num_weights"]
+        else:
+            n_weights = 100
+        weights = [_uniform(len(self._objectives)) for _ in range(n_weights)]
+        weights = [w * self._sign_vector for w in weights]
+        reporter = MODistStatusReporter(self._objectives, weights, 
+                self._scalarization_options, remote=task.resources.node)
+        kwargs["reporter"] = reporter
+        super().add_job(task, **kwargs)
+
+    def get_pareto_front(self):
```

Comment:
It would be good to have this function outside of the scheduler class. Namely, we usually store training_history as results to S3, and then we'd like to compute the Pareto frontier from that, but we do not have a scheduler object then. So, best if this could be something in utils somewhere, and it could then also be a method here.